db_test.go

```diff
@@ -134,6 +134,37 @@ func TestInsert(t *testing.T) {
 	})
 }
 
+func TestSaveExistingDocument(t *testing.T) {
```

Comment:
This function is supposed to test if **Save** is correctly updating an existing document. However, you are inserting a new one